shuup/front/apps/product_filter/views.py

```diff
@@ -0,0 +1,266 @@
+)
+from shuup.core.models import (
+    Attribute,
+    Product,
+    Category,
+    Product,
+)
+from .forms import (
+    BaseFilterSettingsForm,
+    CategoriesFilterSettingsForm,
+    AttributesFilterSettingsForm,
+    ProductFilterForm,
+)
+from .models import (
+    BasicFilterSettingsModel,
+    CategoriesFilterSettingsModel,
+    BasicAttributesFilterSettingsModel,
+    AttributesFilterSettingsModel,
+    BASIC_ATTRIBUTE_FIELDS,
+    LAYOUT_CHOICES,
+)
+
+
+class BaseFilterSettingsFormPart(FormPart):
+    priority = 1
+    name = 'base_settings'
+    form = BaseFilterSettingsForm
+
+    def get_form_defs(self):
+        yield TemplatedFormDef(
+            self.name,
+            self.form,
+            template_name='shuup/product_filter/admin/edit_base_settings_form.jinja',
+            required=True,
+            kwargs={
+                'instance': self.object
+            }
+        )
+
+    def form_valid(self, form):
+        self.object = form[self.name].save()
+        self.object.shop = get_shop(self.request)
+        self.object.save()
+        return self.object
+
+
+class CategoriesFilterSettingsFormPart(FormPart):
+    priority = 2
+    name = 'categories_settings'
+    form = CategoriesFilterSettingsForm
+
+    def get_form_defs(self):
+        try:
+            instance = CategoriesFilterSettingsModel.objects.filter(
+                shop=get_shop(self.request)
+            ).first()
+        except:
+            instance = CategoriesFilterSettingsModel()
+        yield TemplatedFormDef(
+            self.name,
+            self.form,
+            template_name='shuup/product_filter/admin/edit_categories_settings_form.jinja',
+            required=True,
+            kwargs={
+                'instance': instance,
+            }
+        )
+
+    def form_valid(self, form):
+        self.object = form[self.name].save()
+        self.object.shop = get_shop(self.request)
+        self.object.save()
+        return self.object
+
+
+class AttributesFilterSettingsFormPart(FormPart):
+    priority = 3
+    name = 'attributes_settings'
+    form = AttributesFilterSettingsForm
+
+    def get_form_defs(self):
+        try:
+            instance = AttributesFilterSettingsModel.objects.filter(
+                shop=get_shop(self.request)
+            ).first()
+        except AttributesFilterSettingsModel.DoesNotExist:
+            instance = AttributesFilterSettingsModel()
+        yield TemplatedFormDef(
+            self.name,
+            self.form,
+            template_name='shuup/product_filter/admin/edit_attributes_settings_form.jinja',
+            required=True,
+            kwargs={
+                'instance': instance,
+                'initial': self.get_initial(),
+            }
+        )
+
+    def get_initial(self):
+        try:
+            objects = AttributesFilterSettingsModel.objects.filter(
+                shop=get_shop(self.request)
+            )
+        except AttributesFilterSettingsModel.DoesNotExist:
```

Comment:
Filter never throws does not exist. Same applies to rest of the queryset with only filter which returns just empty list when there is no matches.